### PR TITLE
Support pull-from-upstream for non-git upstreams

### LINKED ISSUE #2203: Allow pull from upstream for non-git upstreams
- [ ] use the implementation from packit/packit#2099
- [ ] figure out the DB event handling
   - currently, the [ProjectReleaseModel](https://github.com/packit/packit-service/blob/16236087fb8071c83b3802464f3af504f4fb933e/packit_service/models.py#L1088) is always used
   - the model is added [here](https://github.com/packit/packit-service/blob/16236087fb8071c83b3802464f3af504f4fb933e/packit_service/worker/events/new_hotness.py#L53)
 - [ ] if needed, update the API implementation [here](https://github.com/packit/packit-service/blob/16236087fb8071c83b3802464f3af504f4fb933e/packit_service/service/api/utils.py#L71) and [here](https://github.com/packit/packit-service/blob/16236087fb8071c83b3802464f3af504f4fb933e/packit_service/service/api/pull_from_upstream.py)

*Part of #2202 epic*

### PR DESCRIPTION
**Pull request by lbarcziova**
Fixes #2203 

TODO:

- [ ] dashboard followup

RELEASE NOTES BEGIN

`pull-from-upstream` is now supported also without specifying `upstream_project_url` in the configuration. For such cases, the upstream repository will not be detected and cloned during release syncing.

RELEASE NOTES END

### CODE REVIEW

**Review by majamassarini — APPROVED** [review]
🚀🚀🚀

File: packit_service/service/api/utils.py [hunk 1]
```diff
@@ -89,10 +91,38 @@ def get_sync_release_info(sync_release_model: SyncReleaseModel):
...
+
+    return result_dict
+
+
+def get_project_info(project: Union[AnityaProjectModel, GitProjectModel]):
+    result_dict = {}
+
+    anitya_project_id = anitya_project_name = anitya_package = project_url = (
+        repo_name
+    ) = repo_namespace = None
```
Comment (nforro) [inline packit_service/service/api/utils.py:108]: A nitpick, but this is really ugly :sweat_smile: How about this?

```suggestion
    anitya_project_id = anitya_project_name = anitya_package = None
    project_url = repo_name = repo_namespace = None
```

**Review by nforro — APPROVED** [review]
LGTM

**Review by lachmanfrantisek — APPROVED** [review]
Wow, I was expecting many more changes!

:rocket: 

(edit: forgot to submit this..;)

**Comment by Venefilyn** [comment]
I'm curious, what is Anitya?

**Comment by lbarcziova** [comment]
It is a monitoring platform for upstream projects that checks for new releases, see [docs](https://release-monitoring.org/static/docs/index.html). It is used in Packit for `pull-from-upstream` jobs. [This](https://release-monitoring.org/project/23856/) is how project in Anitya looks like.

**Pull request merged**